### PR TITLE
feat: break out required and optional flags in subcommand help

### PR DESCRIPTION
**Pull request by k3llymariee**
```shell
ldcli teams get --help
```
image

### CODE REVIEW

File: cmd/resources/resources.go [hunk 1]
```diff
@@ -135,6 +135,9 @@ func GetTemplateData(fileName string) (TemplateData, error) {
 			for _, p := range op.Parameters {
 				if p.Value != nil {
 					// TODO: confirm if we only have one type per param b/c somehow this is a slice
+					if strings.Contains(p.Value.Description, "Deprecated") {
+						continue
+					}
```
Comment (k3llymariee) [inline cmd/resources/resources.go:140]: sneaking this in here as i noticed we had some deprecated flags in the cli

File: cmd/resources/resources.go [hunk 2]
```diff
@@ -352,10 +356,49 @@ func NewOperationCmd(parentCmd *cobra.Command, client resources.Client, op Opera
...
+  {{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}{{end}}{{end}}{{end}}{{if .HasAvailableLocalFlags}}
+
+Required flags:
+{{WrappedRequiredFlagUsages . | trimTrailingWhitespaces}}
+
+Optional flags:
+{{WrappedOptionalFlagUsages . | trimTrailingWhitespaces}}
+
+Global Flags:
+{{rpad "  -h, --help" 29}} Help for this command
```
Comment (k3llymariee) [inline cmd/resources/resources.go:396]: it felt weird to include this within the "optional" flags, even though it's not technically a global flag it kinda is? lmk if we want to move this somewhere else

File: cmd/templates.go [hunk 3]
```diff
@@ -21,3 +27,44 @@ Flags:
...
+	width, _, err := term.GetSize(0)
+	if err != nil {
+		width = 80
+	}
+
+	return width
+}
+
+func init() {
+	cobra.AddTemplateFunc("WrappedRequiredFlagUsages", WrappedRequiredFlagUsages)
```
Comment (dbolson) [inline cmd/templates.go:68]: Could you move these to [`Execute()`](https://github.com/launchdarkly/ldcli/blob/main/cmd/root.go#L190) so we don't use `init()`?

**Review by dbolson — APPROVED** [review]

File: cmd/templates.go (hunk 3 above)
Comment (k3llymariee) [reply cmd/templates.go]: oh yup!

**Pull request merged**